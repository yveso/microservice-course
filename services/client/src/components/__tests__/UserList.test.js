import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

import UserList from '../UsersList'

const users = [
  {
    active: true,
    email: 'yves@yves.de',
    id: 1,
    username: 'yves',
  },
  {
    active: true,
    email: 'yves2@yves.de',
    id: 2,
    username: 'yves2',
  },
]

test('UserList renders', () => {
  const wrapper = shallow(<UserList users={users} />);
  const element = wrapper.find('h4');
  expect(element.length).toBe(2);
  expect(element.get(0).props.children).toBe('yves');
})

test('UserList renders snapshot', () => {
  const tree = renderer.create(<UserList users={users}/>).toJSON();
  expect(tree).toMatchSnapshot()
})
