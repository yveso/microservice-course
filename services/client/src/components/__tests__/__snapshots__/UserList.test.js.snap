// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserList renders snapshot 1`] = `
<div>
  <h4
    className="box title is-4"
  >
    yves
  </h4>
  <h4
    className="box title is-4"
  >
    yves2
  </h4>
</div>
`;
